firewall/rule_enforcer.go

```diff
@@ -236,21 +237,35 @@ func (r *RuleEnforcer) handleRequest(ctx context.Context,
 		return nil, fmt.Errorf("error parsing proto: %v", err)
 	}
 
+	var errs []error
 	for _, rule := range rules {
 		newRequest, err := rule.HandleRequest(ctx, ri.URI, msg)
 		if err != nil {
-			st := status.Errorf(
-				codes.ResourceExhausted, "rule violation: %v",
-				err,
-			)
-			return nil, st
+			errs = append(errs, err)
```

Comment:
yes, I can see that argument of resource usage, the current approach enables testing for different rule violations in a single test, see https://github.com/lightninglabs/lightning-terminal/blob/a6b2b260bd036ee9ddfc81054737bde51d3919f4/itest/litd_firewall_test.go#L1104-L1105 and https://github.com/lightninglabs/lightning-terminal/blob/a6b2b260bd036ee9ddfc81054737bde51d3919f4/itest/litd_firewall_test.go#L1320-L1322
It also gives a complete list of errors, otherwise we'd only know the first successful ones, but there could be others after the first failed one. I can change it to your suggested approach, if you think it makes more sense